tests/hash/main.cpp

```diff
@@ -144,6 +153,75 @@ int main(int argc, char *argv[]) {
     }
   }
   input.open(argv[2], ios_base::in);
+
+  if (f == hash_blake2b) {
+    // blake2b does use different format and has key for hashing.
+    while (true) {
+      #define HASH_DATA_LEN 1024
```

Comment:
```suggestion
      static constexpr size_t HASH_DATA_LEN = 1024;
```

This change would limit this variable to the scope of the `while` loop and won't ever cause unexpected preprocessor tokens to be generated later in the file, especially since this name is so generic.